src/Plugin/EntityBrowser/Widget/MediaCreationForm.php

```diff
@@ -0,0 +1,187 @@
+    $target_bundles = $context['widget_context']['target_bundles'] ?? [];
+    $bundles = $this->entityTypeBundleInfo->getBundleInfo('media');
+    if ($target_bundles) {
+      $bundles = array_intersect_key($bundles, $target_bundles);
+    }
+
+    $options = [];
+    foreach ($bundles as $bundle => $info) {
+      $options[$bundle] = $info['label'];
+    }
+
+    $id = Html::getId('media_entity_form');
+
+    $form['media_bundle'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Bundle'),
+      '#options' => $options,
+      '#required' => TRUE,
+      '#ajax' => [
+        'callback' => [$this, 'ajaxCallback'],
+        'wrapper' => $id,
+      ],
+    ];
+
+    $form['entity_form'] = [
+      '#type' => 'container',
+      '#id' => $id,
+    ];
+
+    // Here we need to use the user input because there may be other ajax
+    // requests (such as the one for a file upload) which won't contain the
+    // bundle in the form state values.
+    $user_input = $form_state->getUserInput();
+    $bundle = isset($user_input['media_bundle']) ? $user_input['media_bundle'] : NULL;
+    if ($bundle && isset($options[$bundle])) {
+      // Pretend to be IEFs submit button.
+      $form['#submit'] = [
+        [
+          'Drupal\inline_entity_form\ElementSubmit',
+          'trigger',
+        ],
+      ];
+      $form['actions']['submit']['#ief_submit_trigger'] = TRUE;
+      $form['actions']['submit']['#ief_submit_trigger_all'] = TRUE;
+
+      $form['entity_form']['inline_entity_form'] = [
+        '#type' => 'inline_entity_form',
+        '#op' => 'add',
+        '#entity_type' => 'media',
+        '#bundle' => $bundle,
+        '#form_mode' => 'default',
+      ];
+    }
+
+    return $form;
+  }
+
+  /**
+   * Ajax callback to rebuild the media creation form.
+   *
+   * @param array $form
+   *   The form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   *
+   * @return array
+   *   The form element.
+   */
+  public function ajaxCallback(array &$form, FormStateInterface $form_state): array {
+    return NestedArray::getValue($form, array_merge([
+      $form['#browser_parts']['widget'],
+      'entity_form',
+    ]));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function prepareEntities(array $form, FormStateInterface $form_state) {
+    $entity_form = $form[$form['#browser_parts']['widget']]['entity_form'];
+    if (!isset($entity_form['inline_entity_form'])) {
+      return [];
+    }
+    return [$entity_form['inline_entity_form']['#entity']];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submit(array &$element, array &$form, FormStateInterface $form_state) {
+    if (!empty($form_state->getTriggeringElement()['#eb_widget_main_submit'])) {
+      $entities = $this->prepareEntities($form, $form_state);
+      array_walk(
+        $entities,
+        function (EntityInterface $entity) {
+          $entity->save();
+        }
+      );
```

Comment:
I know this is copied from the other, but let's indent properly
```suggestion
      array_walk($entities, function (EntityInterface $entity) {
        $entity->save();
      });
```